zeppelin-plugins/launcher/docker/src/main/java/org/apache/zeppelin/interpreter/launcher/DockerInterpreterProcess.java

```diff
@@ -0,0 +1,652 @@
+    String tarFile = file2Tar(copyFiles);
+
+    // copy tar to ZEPPELIN_CONTAINER_DIR, auto unzip
+    InputStream inputStream = new FileInputStream(tarFile);
+    try {
+      docker.copyToContainer(inputStream, containerId, CONTAINER_UPLOAD_TAR_DIR);
+    } finally {
+      inputStream.close();
+    }
+
+    // copy all files in CONTAINER_UPLOAD_TAR_DIR to the root directory
+    cpdirInContainer(containerId, CONTAINER_UPLOAD_TAR_DIR + "/*", "/");
+
+    // delete tar file in the local
+    File fileTar = new File(tarFile);
+    fileTar.delete();
+  }
+
+  private void mkdirInContainer(String containerId, String path)
+      throws DockerException, InterruptedException {
+    String execCommand = "mkdir " + path + " -p";
+    execInContainer(containerId, execCommand, true);
+  }
+
+  private void rmInContainer(String containerId, String path)
+      throws DockerException, InterruptedException {
+    String execCommand = "rm " + path + " -R";
+    execInContainer(containerId, execCommand, true);
+  }
+
+  private void cpdirInContainer(String containerId, String from, String to)
+      throws DockerException, InterruptedException {
+    String execCommand = "cp " + from + " " + to + " -R";
+    execInContainer(containerId, execCommand, true);
+  }
+
+  private void execInContainer(String containerId, String execCommand, boolean logout)
+      throws DockerException, InterruptedException {
+
+    LOGGER.info("exec container commmand: " + execCommand);
+
+    final String[] command = {"sh", "-c", execCommand};
+    final ExecCreation execCreation = docker.execCreate(
+        containerId, command, DockerClient.ExecCreateParam.attachStdout(),
+        DockerClient.ExecCreateParam.attachStderr());
+
+    LogStream logStream = docker.execStart(execCreation.id());
+    while (logStream.hasNext() && logout) {
+      final String log = UTF_8.decode(logStream.next().content()).toString();
+      LOGGER.info(log);
+    }
+  }
+
+  private String file2Tar(HashMap<String, String> copyFiles) throws IOException {
+    File tmpDir = Files.createTempDir();
+
+    Date date = new Date();
+    String tarFileName = tmpDir.getPath() + date.getTime() + ".tar";
+
+    List<TarFileEntry> tarFileEntries = new ArrayList<>();
+    for (Map.Entry<String, String> entry : copyFiles.entrySet()) {
+      String filePath = entry.getKey();
+      String archivePath = entry.getValue();
+      TarFileEntry tarFileEntry = new TarFileEntry(new File(filePath), archivePath);
+      tarFileEntries.add(tarFileEntry);
+    }
+
+    TarUtils.compress(tarFileName, tarFileEntries);
+
+    return tarFileName;
+  }
+
+  @VisibleForTesting
+  boolean isSpark() {
+    return "spark".equalsIgnoreCase(interpreterGroupName);
+  }
+
+  private String getZeppelinHome() throws IOException {
+    String zeppelinHome = "";
+    if (System.getenv("ZEPPELIN_HOME") != null) {
+      zeppelinHome = System.getenv("ZEPPELIN_HOME");
+    }
+    if (StringUtils.isBlank(zeppelinHome)) {
```

Comment:
In which case `ZEPPELIN_HOME` will be none ?

Reply:
Just to be more reliable.